pkg/internal/source/event_handler.go

```diff
@@ -43,19 +43,19 @@ func NewEventHandler(ctx context.Context, queue workqueue.RateLimitingInterface,
 }
 
 // EventHandler adapts a handler.EventHandler interface to a cache.ResourceEventHandler interface.
-type EventHandler struct {
+type EventHandler[T client.Object] struct {
```

Comment:
This can also use `EventHandler[T any]` without changes to allow watching non-kubernetes resources.

Reply:
This is internal and only used by `source.Kind` which only works with `client.Object`, so there is no reason to have a different type constraint here